pkg/api/transport.go

```diff
@@ -6,8 +6,12 @@ import (
 	"time"
 )
 
-// NewTransport creates a new http.Transport, with default 30 second TLS timeout.
-func NewTransport() *http.Transport {
```

Comment:
Let's keep the original signature unchanged here. We can achieve the same without changing it. As commented above, this will make the logic simpler, as we do not manipulate a passed in object with a special case for passing in nil.

Reply:
Updated.